openfga/v1/openfga_service.proto

```diff
@@ -237,8 +242,71 @@ service OpenFGAService {
         "  \"authorization_model_id\": \"01G50QVV17PECNVAHX1GG4Y5NC\"\n"
         "}\n"
         "```\n"
-        "OpenFGA's response will include `{ \"allowed\": true }` if there is a relationship "
-        "and `{ \"allowed\": false }` if there isn't."
+        "### Querying usersets\n"
+        "Some Checks will always return `true`, even without any tuples. For example, for the following authorization model\n"
+        "```python\n"
+        "model\n"
+        "  schema 1.1\n"
+        "type user\n"
+        "type document\n"
+        "  relations\n"
+        "    define reader: [user]\n"
+        "```\n"
+        "the following query\n"
+        "```json\n"
+        "{\n"
+        "  \"tuple_key\": {\n"
+        "     \"user\": \"document:2021-budget#reader\",\n"
+        "     \"relation\": \"reader\",\n"
+        "     \"object\": \"document:2021-budget\"\n"
+        "  }\n"
+        "}\n"
+        "```\n"
+        "will always return `{ \"allowed\": true }`. This is because usersets are self-defining: the userset `document:2021-budget#reader` will always have the `reader` relation with `document:2021-budget`.\n"
+        "### Querying usersets with exclusion in the model\n"
+        "A Check for a userset can yield results that must be treated carefully if the model involves exclusion. For example, for the following authorization model\n"
+        "```python\n"
+        "model\n"
+        "  schema 1.1\n"
+        "type user\n"
+        "type group\n"
+        "  relations\n"
+        "    define member: [user]\n"
+        "type document\n"
+        "  relations\n"
+        "    define blocked: [user]\n"
+        "    define reader: [group#member] but not blocked\n"
+        "```\n"
+        "the following query\n"
+        "```json\n"
+        "{\n"
+        "  \"tuple_key\": {\n"
+        "     \"user\": \"group:finance#member\",\n"
+        "     \"relation\": \"reader\",\n"
+        "     \"object\": \"document:2021-budget\"\n"
+        "  },\n"
+        "  \"contextual_tuples\": {\n"
+        "    \"tuple_keys\": [\n"
+        "      {\n"
+        "        \"user\": \"user:anne\",\n"
+        "        \"relation\": \"member\",\n"
+        "        \"object\": \"group:finance\"\n"
+        "      },\n"
+        "      {\n"
+        "        \"user\": \"group:finance#member\",\n"
+        "        \"relation\": \"reader\",\n"
+        "        \"object\": \"document:2021-budget\"\n"
+        "      },\n"
+        "      {\n"
+        "        \"user\": \"user:anne\",\n"
+        "        \"relation\": \"blocked\",\n"
+        "        \"object\": \"document:2021-budget\"\n"
+        "      }\n"
+        "    ]\n"
+        "  },\n"
+        "}\n"
+        "```\n"
+        "will return `{ \"allowed\": true }`, even though a specific user of the userset `group:finance#member` does not have the `reader` relationship with the given object.\n"
```

Comment:
FWIW, this happens on `main` today. https://play.fga.dev/stores/create/?id=01HT1B1QQ96HZ44ZBVEVERX1E1

<img width="1508" alt="image" src="https://github.com/openfga/api/assets/5374887/77c75358-f91e-4939-8c4e-dc414cd8058f">